modules/gdscript/gdscript.cpp

```diff
@@ -2019,6 +2019,50 @@ Error GDScriptLanguage::execute_file(const String &p_path) {
 }
 
 void GDScriptLanguage::finish() {
+	finalize();
+}
+
+void GDScriptLanguage::finalize() {
+	if (finalizing) {
+		return;
+	}
+	finalizing = true;
```

Comment:
This would now work, but I'm a bit confused because now `finalize()` will be called both by `GDScriptLanguage::finish()` and `GDScriptLanguage::~GDScriptLanguage()`, but it will only do something for the first one that calls it.

Is that order non deterministic, or why do we need to call it in both?

Reply:
Mmm. Yeah, calling it once should work. I'll let the `finalizing` variable there, just to be sure that it's not called twice.

Reply:
I guess there's no need for a new `finalize()` method now and it can go in `finish()` directly?

Reply:
Yes. I was inspired by `csharp_script.cpp`, which explains a lot, but yes, I'll transfer everything into `finish()`.

https://github.com/godotengine/godot/blob/bcc061edaeb8676b82dba0d857cfbf66638071b5/modules/mono/csharp_script.cpp#L128-L140

Reply:
Maybe @neikeq can give some input then, there might be a good reason to still call `finalize()` in both `finish()` and the destructor.

Reply:
> Maybe @neikeq can give some input then, there might be a good reason to still call `finalize()` in both `finish()` and the destructor.

I don't really remember, to be honest, and I'm afraid changing it could cause an issue. I wish I had left an explanatory comment there...

The reason it's a different method (`finalize`) is because clang-tidy warns when calling virtual methods (`finish`) from the destructor.